src/tribol/mesh/MfemData.cpp

```diff
@@ -340,46 +377,60 @@ void MfemMeshData::UpdateMfemMeshData( RealT binning_proximity_scale )
     redecomp_elem_thickness_ =
         std::make_unique<mfem::QuadratureFunction>( new mfem::QuadratureSpace( &GetRedecompMesh(), 0 ) );
     redecomp_elem_thickness_->SetOwnsSpace( true );
+    redecomp_elem_thickness_->UseDevice( use_device_ );
     *redecomp_elem_thickness_ = 0.0;
     redecomp_xfer.TransferToSerial( *elem_thickness_, *redecomp_elem_thickness_ );
     // set element thickness on tribol mesh
-    tribol_elem_thickness_1_ = std::make_unique<ArrayT<RealT>>( 0, GetElemMap1().empty() ? 1 : GetElemMap1().size() );
-    for ( auto redecomp_e : GetElemMap1() ) {
-      mfem::Vector quad_val;
-      redecomp_elem_thickness_->GetValues( redecomp_e, quad_val );
-      tribol_elem_thickness_1_->push_back( quad_val[0] );
-    }
-    tribol_elem_thickness_2_ = std::make_unique<ArrayT<RealT>>( 0, GetElemMap2().empty() ? 1 : GetElemMap2().size() );
-    for ( auto redecomp_e : GetElemMap2() ) {
-      mfem::Vector quad_val;
-      redecomp_elem_thickness_->GetValues( redecomp_e, quad_val );
-      tribol_elem_thickness_2_->push_back( quad_val[0] );
-    }
+    tribol_elem_thickness_1_ = std::make_unique<ArrayT<RealT>>(
```

Comment:
I'm moving through this block quickly. It just looks like updating the data types to use the array views?